src/cache/Cache.ts

```diff
@@ -40,25 +41,25 @@ async function fetchSingleRange(
     );
   }
 
-  const { history, range } = historyInRange;
   /*
   home assistant will "invent" a datapoiont at startT with the previous known value, except if there is actually one at startT.
   To avoid these duplicates, the "fetched range" is capped to end at the last known point instead of endT.
   This ensures that the next fetch will start with a duplicate of the last known datapoint, which can then be removed.
   On top of that, in order to ensure that the last known point is extended to endT, I duplicate the last datapoint
   and set its date to endT.
   */
+  /* @TODO: confirm if the following is still relevant.
+      If it is, then HA's invented datapoint needs to be handled, and the completion to Date.now() should be handled
+      while plotting instead of inside the cache (or not handled at all)
+   */
+  let range: [number, number] = [startT, startT];
   if (history.length) {
-    const last = history[history.length - 1];
-    const dup = JSON.parse(JSON.stringify(last));
-    history[0].duplicate_datapoint = true;
-    dup.duplicate_datapoint = true;
-    dup.last_updated = Math.min(+end, Date.now());
-    history.push(dup);
+    range = [startT, history[history.length - 1].timestamp];
+    history[0].fake_boundary_datapoint = true;
```

Comment:
@zanna-37 I was doing something else, but it required me to review the code that handles the invented points.
I had remembered it incorrectly and thought HA sent a fake datapoint at the end of the requested interval. Instead (and I think you mentioned this already), HA sends a fake datapoint at the start of the requested interval.

What I did as part of this WIP PR seems to work really well:
* Fetch from `startT-1`: this way the first datapoint is either fake, or irrelevant for the requested range.
* Mark it for deletion (which will only happen if the datapoint is not the first one in the cache)
* And then (this is below), when clearing old data from the cache, don't modify the first and last datapoints so they match the visible range. Instead, leave them as they are. This is now possible because the visible range is now set explicitely.

Reply:
what do you mean by "clearing old data from the cache"? When does this happen?

Reply:
This happens when one is not "browsing" i.e if one doesn't zoom/scroll or after clicking reset. To avoid accumulating datapoints in ram for ever, the caches is cleared of all data out of the `hours_to_show`.

Reply:
I also changed how the traces are extended to the present: Instead of doing it inside of the cache, the extra datapoint is only created during plotting. This means that there is nothing to manage or remove later, as far as the cache is concerned, those "fake" points don't exist.

Reply:
Right, well done!

Reply:
Note that now that the axis rang is set explicitly, there's no need for it to be this aggressive